gensim/models/word2vec.py

```diff
@@ -562,14 +573,6 @@ def scale_vocab(self, min_count=None, sample=None, dry_run=False):
         # print extra memory estimates
         report_values['memory'] = self.estimate_memory(vocab_size=len(retain_words))
 
-        if not dry_run:
-            new_vocab = {}
-            for w in retain_words:
-                new_vocab[w] = self.vocab[w]
-                new_vocab[w].index = len(self.index2word)
-                self.index2word.append(w)
-            self.vocab = new_vocab
-            logger.info("vocabulary min_count & sample applied, and indexes assigned")
         return report_values
```

Comment:
I was assuming that not dropping `raw_vocab` was an oversight. Yes, if it's kept users will even be able to re-scale up. 

But, holding that extra large dict makes out-of-memory errors *more* likely for people who don't use any `min_count` trimming. I could see making this a user-controlled option, perhaps as another visible option on `scale_vocab` (which is where people would be told to look/tinker if hitting vocab-size related issues). My preference would be for the default to be the "old behavior" (a min_count trim, once done, is irreversible – minimizing risks of OOM) but the new retain-raw-frequencies capability is there for experts/code-reviewers to discover.